sql/src/main/java/org/apache/druid/sql/http/SqlResource.java

```diff
@@ -271,4 +332,14 @@ public Response cancelQuery(
       return Response.status(Status.FORBIDDEN).build();
     }
   }
+
+  @VisibleForTesting
+  String applyErrorMessageFilter(final String errorMessage, final List<Pattern> whitelistRegex)
```

Comment:
Maybe this should be a utility method on `QueryException` or someplace similar so it can be better shared for when JDBC and native JSON queries are added (also a few other places i see this `.stream().noneMatch(pattern -> pattern.matcher(errorMessage).matches()))` pattern in this PR that could probably be shared). If you imagine this will only ever be done to queries, maybe it makes sense to just bake method into `QueryException` so that all implementations just know how to trim the information they contain, then everything just has to make sure they throw some sort of `QueryException` and everything should be good.

Thinking further though, if this functionality only applies to queries, do we really need to use regex? Could we instead just configure a set of exception message by errorCode or maybe the classes of `QueryException` (not errorClass, but like java class name) to allow, and anything that doesn't match would be sanitized? I worry that regex based on the error messages might be a bit tricky/steep learning curve, and if we go this way we should probably update the documents to include several examples of the types of messages that query errors can encounter, and example configs.

Reply:
Moved to a common class so that we can reuse in multiple places.